### PR TITLE
feat: `forecastOuma()`. (#10)

### LINKED ISSUE #10: feat: Implement function `forecasteOuma()`

### PR DESCRIPTION
**Pull request by MinoruSekine**
Closes #10 

## Summary by CodeRabbit

* **New Features**
  * Added Ouma-Irai forecast feature to determine quest and dungeon for given dates.

* **Chores**
  * Project scaffolding and tooling: package manifest, TypeScript config, editor configuration, ESLint rules, and Makefile with build/lint/test targets.
  * Added CI workflows to run linting and tests automatically.

* **Tests**
  * Added unit tests covering forecast behavior and edge cases.

### CODE REVIEW

File: src/ouma_forecaster.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    } else {
+      return { succeeded: false, error_detail: prevOumaIndex.error_detail };
+    }
+  } else {
+    return { succeeded: false, error_detail: "2025 or earlier is not supported." };
+  }
+}
+
+export function forecastOuma(date: Date): OumaForecastResult {
+  const resultIndex = forecastOumaIndex(date);
```
[the remark this answers is not in the document]
Comment (MinoruSekine) [reply src/ouma_forecaster.ts:110]: `forecastOumaIndex()` doesn't expect normalized `Date` input.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  eslint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version-from-package-json: "volta.node"
+          cache: 'npm'
```
[the remark this answers is not in the document]
Comment (MinoruSekine) [reply .github/workflows/test.yml:18]: Replaced it into `node-version-file: package.json`.

**Comment by MinoruSekine** [comment]
@coderabbitai review

File: src/ouma_forecaster.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      const numQuestsInDungeon = oumaMap[prevOumaIndex.dungeon][1].length;
+      const quest_index = (prevOumaIndex.quest + 1) % numQuestsInDungeon;
+      return { succeeded: true, dungeon: prevOumaIndex.dungeon, quest: quest_index };
+    } else {
+      return { succeeded: false, error_detail: prevOumaIndex.error_detail };
+    }
+  } else {
+    return { succeeded: false, error_detail: "2025 or earlier is not supported." };
+  }
+}
```
[the remark this answers is not in the document]
Comment (MinoruSekine) [reply src/ouma_forecaster.ts:107]: Updating `oumaDict` can suppress recursion depth exhausting,
this recursion makes implementations' simplicity.

File: test/ouma_forecaster.test.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
+/**
+ * @overview Tests for forecasting quest of "Ouma-Irai"
+ *           in Nobunaga's ambition Online.
+ *
+ * @author Minoru Sekine
+ * @copyright Copyrght 2026 Minoru Sekine
```
[the remark this answers is not in the document]
Comment (MinoruSekine) [reply test/ouma_forecaster.test.ts]: This has been resolved.
